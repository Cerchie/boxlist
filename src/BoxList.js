import React, {useState} from 'react';

import Box from './Box'
import NewBoxForm from './NewBoxForm'

import {v4 as uuid} from 'uuid';

const BoxList = () => {
const INITIAL_STATE = [
    {id: uuid(), width: '200', height: '80', backgroundColor: 'red'},
]
const [boxes, setBoxes] = useState(INITIAL_STATE)

const addBox = (newBox) => {
setBoxes(boxes => [...boxes, {...newBox, id: uuid() }])
}
const remove = id => {
    setBoxes(boxes => boxes.filter(box => box.id !== id));
  };
return (
    <div>
        <h3>Box List</h3>
        <NewBoxForm addBox={addBox} />
        <div>
            {boxes.map((box) => <Box 
            key={box.id} 
            id={box.id} 
            backgroundColor={box.backgroundColor} 
            width={box.width} 
            height={box.height} 
            remove={remove}/>)}
        </div>
    </div>
)
}
export default BoxList;