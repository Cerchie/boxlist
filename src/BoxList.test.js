import React from 'react';
import {render, fireEvent } from "@testing-library/react"
import BoxList from './BoxList'

it("renders without crashing", function(){
    render(<BoxList />)
})

it("matches snapshot", function() {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot(); })

it('should add new item', function () {
    const {queryByText, getByLabelText} = render(<BoxList />)
    const input = getByLabelText('backgroundColor');
    const btn =  queryByText('Add Box')
    expect(queryByText('backgroundColor: Lavendar')).not.toBeInTheDocument();
    fireEvent.change(input, {target: {value: 'Lavendar'}})
    fireEvent.click(btn);
    expect(queryByText('backgroundColor: Lavendar')).toBeInTheDocument();
})