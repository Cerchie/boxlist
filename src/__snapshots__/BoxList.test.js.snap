// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Box List
    </h3>
    <form
      id="form"
    >
      <label
        for="backgroundColor"
      >
         backgroundColor
      </label>
      <input
        name="backgroundColor"
        placeholder="BG Color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
         width: 
      </label>
      <input
        id="width"
        max="900"
        min="10"
        name="width"
        type="range"
        value=""
      />
      <label
        for="height"
      >
        height: 
      </label>
      <input
        id="height"
        max="900"
        min="10"
        name="height"
        type="range"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div>
      <div
        style="background-color: red; width: 200px; height: 80px;"
      >
        COLORBOX
      </div>
      <button
        id="delete-btn"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
